MonoDevelop.FSharpBinding/FSharpSymbolHelper.fs

```diff
@@ -357,11 +364,125 @@ module SymbolTooltips =
+                    print [ asUserType (c.MemberReturnType.Format displayContext) ]
+                else 
+                    if c.MemberArgumentTypes.Count <= 1 then
+                        print [ asUserType "unit" ]
+                    else
+                        printParamName()
+                    print [ asSymbol " -> "
+                            asUserType ((c.MemberReturnType.Format displayContext).TrimStart()) ]
+
+                print [ asBrackets ")" ]
+
+            let typeConstraint (tc: FSharpType) =
+                print [ asSymbol " :> "
+                        asUserType (tc.Format displayContext) ]
+
+            let constructorConstraint () =
+                print [ asSymbol " : "
+                        asBrackets "("
+                        asKeyword "new"
+                        asSymbol " : "
+                        asKeyword "unit"
+                        asSymbol " -> '"
+                        param.DisplayName
+                        asBrackets ")" ]
+
+            let enumConstraint (ec: FSharpType) =
+                print [ asSymbol " : "
+                        asKeyword "enum"
+                        asBrackets (escapeText "<")
+                        asUserType (ec.Format displayContext)
+                        asBrackets (escapeText ">") ]
+
+            let delegateConstraint (tc: FSharpGenericParameterDelegateConstraint) =
+                print [ asSymbol " : "
+                        asKeyword "delegate"
+                        asBrackets (escapeText "<")
+                        asUserType (tc.DelegateTupledArgumentType.Format displayContext)
+                        asSymbol ", "
+                        asUserType (tc.DelegateReturnType.Format displayContext)
+                        asBrackets (escapeText ">") ]
+
+            match constrainedBy with
+            | _ when constrainedBy.IsCoercesToConstraint -> typeConstraint constrainedBy.CoercesToTarget
+            | _ when constrainedBy.IsMemberConstraint -> memberConstraint constrainedBy.MemberConstraintData
+            | _ when constrainedBy.IsSupportsNullConstraint -> print [(asSymbol " : "); (asKeyword "null")]
+            | _ when constrainedBy.IsRequiresDefaultConstructorConstraint -> constructorConstraint()
+            | _ when constrainedBy.IsReferenceTypeConstraint -> print [(asSymbol " : "); asKeyword "not struct"]
+            | _ when constrainedBy.IsEnumConstraint -> enumConstraint constrainedBy.EnumConstraintTarget
+            | _ when constrainedBy.IsComparisonConstraint -> print [(asSymbol " : "); (asKeyword "comparison")]
+            | _ when constrainedBy.IsEqualityConstraint -> print [(asSymbol " : "); (asKeyword "equality")]
+            | _ when constrainedBy.IsDelegateConstraint -> delegateConstraint constrainedBy.DelegateConstraintData
+            | _ when constrainedBy.IsUnmanagedConstraint -> print [(asSymbol " : "); (asKeyword "unmanaged")]
+            | _ -> ()
```

Comment:
Missing IsDefaultsToConstraint, IsNonNullableValueTypeConstraint and IsSimpleChoiceConstraint. 

I never noticed this before - I just tried to get the examples on [msdn](https://msdn.microsoft.com/en-us/library/dd233203.aspx) passing